hardware/CurrentCostMeterTCP.h

```diff
@@ -2,13 +2,13 @@
 
 #include "CurrentCostMeterBase.h"
 
-class CurrentCostMeterTCP: public CurrentCostMeterBase
+class CurrentCostMeterTCP : public CurrentCostMeterBase
 {
-public:
-	CurrentCostMeterTCP(const int ID, const std::string &IPAddress, const unsigned short usIPPort);
+      public:
+	CurrentCostMeterTCP(int ID, const std::string &IPAddress, unsigned short usIPPort);
```

Comment:
Maybe I am missing something, but what is wrong declaring a function parameter with const ?

Reply:
From https://clang.llvm.org/extra/clang-tidy/checks/readability-const-return-type.html

Such use of const is usually superfluous, and can prevent valuable compiler optimizations.

Reply:
Thats for "const returns", I'm referring to the use of 'const' in the function parameters

CurrentCostMeterTCP(const int ID, const std::string &IPAddress, const unsigned short usIPPort);

why could we not declare 'int ID' as const ? or 'usIPPort' ?

Reply:
Ah. http://clang.llvm.org/extra/clang-tidy/checks/readability-avoid-const-params-in-decls.html

Looks like there's no real need for it.